mesonbuild/backend/backends.py

```diff
@@ -1309,6 +1293,22 @@ def create_test_serialisation(self, tests: T.List['Test']) -> T.List[TestSeriali
                 else:
                     raise MesonException('Bad object in test command.')
 
+            is_cross = self.environment.is_cross_build(test_for_machine)
+            exe_wrapper = self.environment.get_exe_wrapper()
+            machine = self.environment.machines[exe.for_machine]
+            if machine.is_windows() or machine.is_cygwin():
+                extra_bdeps: T.List[T.Union[build.BuildTarget, build.CustomTarget, build.CustomTargetIndex]] = []
+                if isinstance(exe, build.CustomTarget):
+                    extra_bdeps = list(exe.get_transitive_build_target_deps())
+                extra_paths = self.determine_windows_extra_paths(exe, extra_bdeps)
```

Comment:
I may be wrong, but I think you could simply do:
`extra_paths = self.determine_windows_extra_paths(exe, extra_bdeps + depends)` intead of the for loop over depends, since `determine_windows_extra_paths` already deduplicate its arguments and tests for BuildTarget.

Reply:
Thanks, that works!